### ISSUE TITLE
kpAddLabels does not draw labels on the right side

### ISSUE THREAD

**Issue by jromanowska** [bernatgel/karyoploteR#112]
I wanted to add a label on the right side of a zoomed plot and it doesn't work with the default settings. I've looked into the source code and it seems like the x coordinate is calculated to be far too large.

Here's a reproducible example:

```
library(GenomicRanges)

zoom_region <- makeGRangesFromDataFrame(
  data.frame(
    seqnames = "chrX",
    start = 152955947,
    end = 153061187
  )
)

kp_tmp <- karyoploteR::plotKaryotype(
  chromosome = "chrX",
  zoom = zoom_region,
  plot.type = 2
)
karyoploteR::kpAddBaseNumbers(
  kp_tmp,
  tick.dist = 10000,
  tick.len = 10,
  minor.ticks = TRUE,
  minor.tick.dist = 2000,
  minor.tick.len = 5,
  cex = 1,
  add.units = TRUE,
  digits = 4
)

# DOESN'T WORK:
karyoploteR::kpAddLabels(
  kp_tmp,
  labels = "model no.",
  data.panel = 2,
  side = "right",
  r0 = 0.1,
  r1 = 0.2,
  col = "darkgreen"
)

# WORKS:
karyoploteR::kpAddLabels(
  kp_tmp,
  labels = "model no.",
  data.panel = 2,
  side = "left", # default side
  r0 = 0.1,
  r1 = 0.2,
  col = "darkgreen"
)

# WORKS:
karyoploteR::kpAddLabels(
  kp_tmp,
  labels = "model no.",
  data.panel = 2,
  label.margin = -18, # margin set manually to the x-coord calculated in kpAddLabels
  side = "right",
  r0 = 0.1,
  r1 = 0.2,
  col = "darkgreen"
)
```

**Comment by bernatgel** [comment]
Hi @jromanowska 

Thanks for reporting this. Yes, this is a known limitation, but I think it should be fairly easy to fix. I'll take a look at it and get back to you once this is done. 

Bernat

**Comment by bernatgel** [comment]
Hi @jromanowska 

This has been fixed in 1.20.3 (Current Bioconductor release) and 1.21.3 (devel Bioconductor). It's accessible now in github and will be available in Bioconductor soon (up to 48 hours aprox).

Thanks for reporting the problem and providing working code! If you find anything else, please report it too!

Bernat
